client/src/utils/fetch.js

```diff
@@ -120,11 +117,34 @@ export const getSiteConfig = async () => {
 };
 
 export const checkout = async payload => {
+  const { API_URL } = getConfig();
+  let checkoutStatus;
+  let errors;
+
   if (payload?.items?.length) {
-    return _postAPI('checkout/');
+    try {
+      // Retrieve csrf token from server
+      const token = await _getAPI('csrf_token');
+
+      // Submit form payload and pass back csrf token
+      const response = await fetch(`${API_URL}/checkout`, {
+        method: 'POST',
+        headers: { 'X-CSRFToken': token.csrfToken },
+        body: JSON.stringify(payload),
+        ...baseRequest,
+      });
+      checkoutStatus = await response.json();
+    } catch (error) {
+      errors = [error];
```

Comment:
Repeat: My discussion item about 4xx and 5xx errors above.

Reply:
https://github.com/GoogleCloudPlatform/avocano/issues/339